guides/rules/no-attrs-snapshot.md

```diff
@@ -0,0 +1,21 @@
+# No `attrs` snapshots
+
+In 2.0.0 [`didRecieveAttrs`](https://guides.emberjs.com/v2.9.0/components/the-component-lifecycle/#toc_formatting-component-attributes-with-code-didreceiveattrs-code) and [`didUpdateAttrs`](https://guides.emberjs.com/v2.9.0/components/the-component-lifecycle/#toc_resetting-presentation-state-on-attribute-change-with-code-didupdateattrs-code) hooks were introduced are called whenever the arguments to a component referencially change. These hooks do recieve params to them, however one should not use them as they force those objects to reify which can be very costly when you have a lot of components on the page. These arguments are also purposly undocumented. If for some reason you need to do a comparison of arguments we suggest that you simply keep a cache on the component.
```

Comment:
Just a side comment, but I suspect most people don't understand what _reify_ means. It may be worth linking or explaining. Reification is more abstract (no pun intended) than it is in statically typed languages, such as Java and C#, where generics are a simple and clear example.